expo/expo
# [go][ios][android] Fix recent project feature with new manifest wrappers

# Why

In https://github.com/expo/expo/pull/12631 (ios) and https://github.com/expo/expo/pull/12600 (android) I missed a few places where the old rawJson was implicitly serialized for sending either in a map over the bridge or into a bundle. This PR addresses the few places that were missed.

# How

- On android, use the compiler to override `toString` and find callsites. Also add an exception for calling `toString` at runtime so that any other offenses are more easily noticed and caught.
- On ios we don't have quite the same ability to find all the calls that would serialize it into a string/json, but most of the same callsites applied.

# Test Plan

On both platforms:
1. Open a project
2. Navigate back to home screen in Expo Go and see recently launched projects section has the project
